hera_mc/cm_handling.py

```diff
@@ -453,168 +112,173 @@ def get_part_from_hpnrev(self, hpn, rev):
         -------
         Part object
         """
-        return self.session.query(PC.Parts).filter(
-            (func.upper(PC.Parts.hpn) == hpn.upper())
-            & (func.upper(PC.Parts.hpn_rev) == rev.upper())).first()
+        return self.session.query(partconn.Parts).filter(
+            (func.upper(partconn.Parts.hpn) == hpn.upper())
+            & (func.upper(partconn.Parts.hpn_rev) == rev.upper())).first()
 
-    def get_rev_part_dictionary(self, hpn, rev, at_date, exact_match):
+    def _get_hpn_list(self, hpn, rev, active, exact_match):
         """
-        This gets the list of hpn that match the rev -- the resulting dictionary
-        is used to get the part and connection "dossiers"
+        Returns hpn,rev zip list to accommodate non-exact matches
 
         Parameters
         ----------
-        hpn : list, str
-            The input hera part number [string or list-of-strings] (whole or first part thereof)
-        rev : list, str
-            Specific revision(s) or category(ies) ('LAST', 'ACTIVE', 'ALL', 'FULL')
-            if list must match length of hpn
-        at_date : str, int
-            Reference date of dossier [something get_astropytime can handle]
-        exact_match :  bool
-            If True, will enforce full part number match
+        hpn : str, list
+            As supplied to get_dossier
+        rev : str, list, None
+            As supplied to get_dossier
+
+        active : class ActiveData
+            Contains the active at appropriate date
+        exact_match : bool
+            Flag to enforce exact match, or first letter
 
         Returns
         -------
-        dict
-            Dictionary containing the revision part information.
-        """
+        zip class
+            Contains the hpn, rev pairs
+        """
+        match_list = cm_utils.match_list(hpn, rev, upper=True)
+        if not exact_match:
+            all_hpn = []
+            all_rev = []
+            for h_hpn, h_rev in match_list:
+                for key in active.parts.keys():
+                    if key.upper().startswith(h_hpn):
+                        k_hpn, k_rev = cm_utils.split_part_key(key)
+                        all_hpn.append(k_hpn.upper())
+                        all_rev.append(h_rev)
+            match_list = cm_utils.match_list(all_hpn, all_rev, upper=True)
+        return match_list
+
+    def _set_ports(self, inp):
+        """
+        Handles 3 distinct phases of get_dossier:  initialize, get ports, set allowed
```

Comment:
Since that these three phases do not share code and return immediately (rather than going on to another phase internally) shouldn't they be separate methods?

Reply:
I went back and forth on this, and decided that since it was a private method it was ok to keep them bundled (which I preferred for future understandability, on my part at least).

Reply:
ok.